### PR TITLE
Migrate langdetect to langid.py with lazy loading (#1208)

### LINKED ISSUE #1208: migrate langdetect to langid.py; avoid needless loading
langdetect offers slower and less accurate support for a smaller range of languages than [langid.py](https://aclanthology.org/P12-3005.pdf)

see e.g. pg 28 of https://aclanthology.org/P12-3005.pdf

proposal:

* replace `langdetect` dep with `langid`
* update `garak/langproviders/base.py` to match `langid` interface
* don't load this unless required (inside `garak.langproviders.base.is_meaning_string()`)

### PR DESCRIPTION
**Pull request by nethum529**
Migrates the language-detection dependency used by the language service from `langdetect` to [`langid.py`](https://aclanthology.org/P12-3005.pdf), and loads it lazily. Closes #1208.

`langid.py` provides faster and more accurate detection across a wider range of languages than `langdetect`, and importing it inside `is_meaning_string()` means the model is only loaded when language detection is actually exercised (i.e. reverse-translation judging) rather than on every import of `garak.langproviders.base`.

### What changed

- `garak/langproviders/base.py`
  - Drop the module-level `from langdetect import ...` (and the now-unused `import logging`).
  - `is_meaning_string()` now uses a lazily-imported `langid.classify()`.
  - Behaviour is preserved: `langdetect` previously raised `LangDetectException` on inputs with no linguistic features (e.g. pure digits/punctuation), which the old code treated as "not meaningful". `langid` never raises, so that path is replicated with an explicit "no alphabetic character → not meaningful" guard. The cheap length/repetition checks now run first so trivial inputs short-circuit before any model load.
- `pyproject.toml` / `requirements.txt`: swap `langdetect==1.0.9` → `langid==1.1.6`.
- `tests/langservice/test_langprovision.py`: add unit tests for `is_meaning_string` (the function previously had none), covering non-English (→ translatable), English, too-short, repetitive, and no-feature inputs, plus a structural test asserting `langdetect` is gone and `langid` is imported lazily.

## Verification

- [x] `python -m pytest tests/langservice/test_langprovision.py -k "is_meaning_string or langid"` → 14 passed
- [x] `python -m pytest tests/test_reqs.py` → passed (requirements.txt / pyproject.toml stay consistent)
- [x] Full suite collects cleanly with `langdetect` uninstalled (no other module imported it): `python -m pytest --co` → 4664 tests, exit 0
- [x] `black --check garak/langproviders/base.py tests/langservice/test_langprovision.py` → clean
- [x] **Verify** non-English text is still flagged meaningful and English / no-feature text is skipped
- [x] **Verify** `langid` is not imported at module load, only when `is_meaning_string()` runs

### CODE REVIEW

File: garak/langproviders/base.py [hunk 1]
```diff
@@ -104,20 +102,21 @@ def contains_invisible_unicode(text: str) -> bool:
 
 def is_meaning_string(text: str) -> bool:
     """Check if the input text is a meaningless sequence or invalid for translation."""
```
Comment (leondz) [inline garak/langproviders/base.py:104]: we should rename function and change this comment - it's too cryptic

suggest "check the input text is suitable for translation"

and `def is_suitable_for_translation`

File: tests/langservice/test_langprovision.py [hunk 2]
```diff
@@ -17,6 +17,52 @@ def test_split_input_text():
...
+    "text",
+    [
+        "Bonjour, ceci est une phrase en français.",
+        "これは日本語の文章です。",
+        "Hola mundo, esto es una frase en español.",
+    ],
+)
+def test_is_meaning_string_true_for_non_english(text):
+    # Non-English, translatable text should be flagged as meaningful
+    assert is_meaning_string(text) is True
```
Comment (leondz) [inline tests/langservice/test_langprovision.py:30]: good, i like that we have coverage here

File: tests/langservice/test_langprovision.py [hunk 3]
```diff
@@ -17,6 +17,52 @@ def test_split_input_text():
...
+
+    from garak.langproviders import base
+
+    module_source = inspect.getsource(base)
+    assert "langdetect" not in module_source
+    module_top = module_source.split("\ndef ")[0]
+    assert "import langid" not in module_top
+    assert "import langid" in inspect.getsource(base.is_meaning_string)
+
+
```
Comment (leondz) [inline tests/langservice/test_langprovision.py:65]: this is a bit too in the weeds and locks us in to a specific implementation pattern, i think it should be dropped

**Review by leondz — COMMENTED** [review]
couple of minor changes, i like it in general
